File: python/Baekjoon/3273.py
import sys

n = int(sys.stdin.readline().rstrip())
arr = list(map(int, sys.stdin.readline().rstrip().split()))
x = int(sys.stdin.readline().rstrip())

arr.sort()
start = 0
end = -1
count = 0

while start - end < n:
    sumNum = arr[start] + arr[end]
    if sumNum == x:
        count += 1
        start += 1
    elif sumNum > x:
        end -= 1
    else: start += 1

print(count)

File: Baekjoon/3273.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const [n, nums, x] = fs.readFileSync(filePath).toString().trim().split("\n");
const numsArr = nums
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);
let count = 0;
let start = 0;
let end = +n - 1;

while (start < end) {
  const tmpSum = numsArr[start] + numsArr[end];
  if (tmpSum == +x) {
    count++;
    start++;
    end--;
  } else if (tmpSum < +x) start++;
  else end--;
}

console.log(count);
